roachtest: tpce/c=5000/nodes=3 failed [nil pointer dereference]

roachtest.tpce/c=5000/nodes=3 [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4549874&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4549874&tab=artifacts#/tpce/c=5000/nodes=3) on master @ [0069e980d756ca580167eb321ac6b348b35313f0](https://github.com/cockroachdb/cockroach/commits/0069e980d756ca580167eb321ac6b348b35313f0):


```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /artifacts/tpce/c=5000/nodes=3/run_1
	tpce.go:97,monitor.go:105,errgroup.go:57: UNCLASSIFIED_PROBLEM: context canceled

	monitor.go:127,tpce.go:105,tpce.go:122,test_runner.go:866: monitor failure: monitor command failure: unexpected node event: 3: dead (exit status 134)
		(1) attached stack trace
		  -- stack trace:
		  | main.(*monitorImpl).WaitE
		  | 	main/pkg/cmd/roachtest/monitor.go:115
		  | main.(*monitorImpl).Wait
		  | 	main/pkg/cmd/roachtest/monitor.go:123
		  | github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests.registerTPCE.func1
		  | 	github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests/tpce.go:105
		  | github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests.registerTPCE.func2
		  | 	github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests/tpce.go:122
		  | [...repeated from below...]
		Wraps: (2) monitor failure
		Wraps: (3) attached stack trace
		  -- stack trace:
		  | main.(*monitorImpl).wait.func3
		  | 	main/pkg/cmd/roachtest/monitor.go:202
		  | runtime.goexit
		  | 	GOROOT/src/runtime/asm_amd64.s:1581
		Wraps: (4) monitor command failure
		Wraps: (5) unexpected node event: 3: dead (exit status 134)
		Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *withstack.withStack (4) *errutil.withPrefix (5) *errors.errorString
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
/cc @cockroachdb/kv-triage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*tpce/c=5000/nodes=3.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-13710

Span after finish panic. I see `replicationStatsVisitor` in the stack trace, is this #76430 @Santamaura?

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: use of Span after Finish. Span: stats-reporter. Finish previously called at: <stack not captured. Set debugUseAfterFinish>
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x34794d5]

goroutine 221 [running]:
panic({0x4508a20, 0xc010cd0970})
	GOROOT/src/runtime/panic.go:1147 +0x3a8 fp=0xc00e958ac8 sp=0xc00e958a08 pc=0x48a768
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).detectUseAfterFinish(0x98a23c0)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:185 +0x134 fp=0xc00e958b30 sp=0xc00e958ac8 pc=0xb92374
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).RecordingType(0xc002537500)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:387 +0x1e fp=0xc00e958b48 sp=0xc00e958b30 pc=0xb92c1e
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).IsVerbose(...)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:395
github.com/cockroachdb/cockroach/pkg/util/log.getSpanOrEventLog({0x61c9098, 0xc002479cb0})
	github.com/cockroachdb/cockroach/pkg/util/log/trace.go:92 +0x3a fp=0xc00e958b78 sp=0xc00e958b48 pc=0xcbbefa
github.com/cockroachdb/cockroach/pkg/util/log.logfDepth({0x61c9098, 0xc002479cb0}, 0x2, 0x3, 0x1, {0x4db763e, 0x19}, {0xc00e958f08, 0x1, 0x1})
	github.com/cockroachdb/cockroach/pkg/util/log/channels.go:55 +0x2a8 fp=0xc00e958de8 sp=0xc00e958b78 pc=0xca0548
github.com/cockroachdb/cockroach/pkg/util/log.shoutfDepth({0x61c9098, 0xc002479cb0}, 0x1, 0x3, 0x0, {0x4db763e, 0x19}, {0xc00e958f08, 0x1, 0x1})
	github.com/cockroachdb/cockroach/pkg/util/log/channels.go:84 +0x28d fp=0xc00e958ea8 sp=0xc00e958de8 pc=0xca092d
github.com/cockroachdb/cockroach/pkg/util/log.loggerOps.Shoutf(...)
	github.com/cockroachdb/cockroach/bazel-out/k8-opt/bin/pkg/util/log/log_channels_generated.go:1422
github.com/cockroachdb/cockroach/pkg/util/log/logcrash.ReportPanic({0x61c9098, 0xc002479cb0}, 0xc00e958f80, {0x4698600, 0x8e9c7b0}, 0xc002537500)
	github.com/cockroachdb/cockroach/pkg/util/log/logcrash/crash_reporting.go:174 +0xc5 fp=0xc00e958f28 sp=0xc00e958ea8 pc=0x117fd45
github.com/cockroachdb/cockroach/pkg/util/log/logcrash.ReportPanicWithGlobalSettings({0x61c9098, 0xc002479cb0}, {0x4698600, 0x8e9c7b0}, 0x91425b)
	github.com/cockroachdb/cockroach/pkg/util/log/logcrash/crash_reporting.go:114 +0x97 fp=0xc00e958f78 sp=0xc00e958f28 pc=0x117fa37
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Recover(0xc0006fad80, {0x61c9098, 0xc002479cb0})
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:250 +0x85 fp=0xc00e958fc0 sp=0xc00e958f78 pc=0x1188225
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2Â·dwrapÂ·10()
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:484 +0x2e fp=0xc00e958fe8 sp=0xc00e958fc0 pc=0x11896ee
runtime.deferCallSave(0xc00e9590f0, 0xc00e959f90)
	GOROOT/src/runtime/panic.go:950 +0x82 fp=0xc00e958ff8 sp=0xc00e958fe8 pc=0x48a362
runtime.runOpenDeferFrame(0xc0004b65f0, 0xc000ba1f40)
	GOROOT/src/runtime/panic.go:889 +0x27b fp=0xc00e959078 sp=0xc00e958ff8 pc=0x489cfb
panic({0x4698600, 0x8e9c7b0})
	GOROOT/src/runtime/panic.go:1038 +0x215 fp=0xc00e959138 sp=0xc00e959078 pc=0x48a5d5
runtime.panicmem(...)
	GOROOT/src/runtime/panic.go:221
runtime.sigpanic()
	GOROOT/src/runtime/signal_unix.go:735 +0x327 fp=0xc00e959188 sp=0xc00e959138 pc=0x4a1187
github.com/cockroachdb/cockroach/pkg/kv/kvserver/reports.(*replicationStatsVisitor).visitNewZone.func2({0xc00a937b00, 0x439}, 0x480, {0x58995ca7, 0x16db5f5c})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/reports/replication_stats_report.go:373 +0x35 fp=0xc00e959198 sp=0xc00e959188 pc=0x34794d5
github.com/cockroachdb/cockroach/pkg/kv/kvserver/reports.visitAncestors({0x61c9098, 0xc0016f9140}, 0x0, 0x0, 0xc00e959498)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/reports/reporter.go:475 +0x163 fp=0xc00e959228 sp=0xc00e959198 pc=0x347bc23
github.com/cockroachdb/cockroach/pkg/kv/kvserver/reports.visitZones({0x61c9098, 0xc0016f9140}, 0x0, 0x0, 0x0, 0xc00e959498)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/reports/reporter.go:437 +0x1f3 fp=0xc00e9593f0 sp=0xc00e959228 pc=0x347ba13
github.com/cockroachdb/cockroach/pkg/kv/kvserver/reports.(*replicationStatsVisitor).visitNewZone(0xc00e4f45d0, {0x61c9098, 0xc0016f9140}, 0xc00e9d0e00)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/reports/replication_stats_report.go:350 +0x12f fp=0xc00e9594d0 sp=0xc00e9593f0 pc=0x34792af
github.com/cockroachdb/cockroach/pkg/kv/kvserver/reports.visitRanges({0x61c9098, 0xc0016f9140}, {0x6147370, 0xc00e4b3cc0}, 0xc0027b80b0, {0xc00e9598c0, 0x3, 0x3})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/reports/reporter.go:611 +0x518 fp=0xc00e9596c8 sp=0xc00e9594d0 pc=0x347c718
github.com/cockroachdb/cockroach/pkg/kv/kvserver/reports.(*Reporter).update(0xc0003f8460, {0x61c9098, 0xc0016f9140}, 0x0, 0x0, 0x1)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/reports/reporter.go:231 +0x8cf fp=0xc00e959db0 sp=0xc00e9596c8 pc=0x347aacf
github.com/cockroachdb/cockroach/pkg/kv/kvserver/reports.(*Reporter).Start.func2({0x61c9098, 0xc002479cb0})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/reports/reporter.go:150 +0x1db fp=0xc00e959f28 sp=0xc00e959db0 pc=0x3479e7b
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2()
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:494 +0x16f fp=0xc00e959fe0 sp=0xc00e959f28 pc=0x11894cf
runtime.goexit()
	GOROOT/src/runtime/asm_amd64.s:1581 +0x1 fp=0xc00e959fe8 sp=0xc00e959fe0 pc=0x4bf181
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:483 +0x445
```

--- a/pkg/kv/kvserver/reports/replication_stats_report.go
+++ b/pkg/kv/kvserver/reports/replication_stats_report.go
@@ -370,7 +370,7 @@ func (v *replicationStatsVisitor) visitNewZone(
 				desiredNumVoters = int(*zone.NumVoters)
 				return true
 			}
-			if *zone.NumReplicas != 0 && desiredNumVoters == 0 {
+			if zone.NumReplicas != nil && desiredNumVoters == 0 {
 				desiredNumVoters = int(*zone.NumReplicas)
 			}
 			// We had already found the zone to report to, but we're haven't found

